src/setup_payload/SetupPayloadHelper.cpp

```diff
@@ -143,6 +143,11 @@ static CHIP_ERROR addParameter(SetupPayload & setupPayload, const SetupPayloadPa
 
 CHIP_ERROR loadPayloadFromFile(SetupPayload & setupPayload, const std::string & filePath)
 {
+#if defined(_LIBCPP_HAS_FILESYSTEM) && _LIBCPP_HAS_FILESYSTEM == 0
```

Comment:
This seems to stub things out at the lowest level.

How about instead just removing the `SetupPayloadHelper` from the build system instead? As it seems impossible to `generateQRCodeFromFilePath` and `generateManualCodeFromFilePath` when there is no file system ability.

Stubbing here seems to make the methods globally available and just not work.

Reply:
> Stubbing here seems to make the methods globally available and just not work.

That's what the project has been doing the whole time, and I didn't know if there was a reason for that or not, so I went for the least invasive patch. I can rework it to move this file out, but should the file itself then also be moved?

Reply:
Probably different source set at least, so people can opt in to use it. I am unsure about the blast radius ... hoping not large and only used when a filesystem actually exists.